### PR TITLE
Fix test_vnet_vxlan 3k failing because arp_responder was not running, added sleep

### PR DESCRIPTION
**Pull request by andriyz-nv**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [x] 201911

#### What is the motivation for this PR?
Fix for test_vnet_vxlan 3k when arp_responer was started but not fully functioning. Sleep ensure that during the packet was send it will be catched on VM side
#### How did you verify/test it?
FIxed issue that packet was send FromServer but not catched on FromVm side

  "stderr": "WARNING: No route found for IPv6 destination :: (no default route?)\n/usr/local/lib/python2.7/dist-packages/paramiko/transport.py:33: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in the next release.\n  from cryptography.hazmat.backends import default_backend\nvnet_vxlan.VNET ... FAIL\n\n======================================================================\nFAIL: vnet_vxlan.VNET\n----------------------------------------------------------------------\nTraceback (most recent call last):\n  File \"ptftests/vnet_vxlan.py\", line 325, in runTest\n    self.FromVM(test)\n  File \"ptftests/vnet_vxlan.py\", line 398, in FromVM\n    verify_packet(self, exp_pkt, test['port'])\n  File \"/usr/lib/python2.7/dist-packages/ptf/testutils.py\", line 2405, in verify_packet\n    % (device, port, result.format()))\nAssertionError: Expected packet was not received on device 0, port 2.\n========== EXPECTED ==========\ndst        : DestMACField         = '24:8a:07:ab:9f:02' (None)\nsrc        : SourceMACField       = u'98:03:9b:f3:f5:80' (None)\ntype       : XShortEnumField      = 33024           (0)\n--\nprio       : BitField             = 0               (0)\nid         : BitField             = 0               (0)\nvlan       : BitField             = 3001            (1)\ntype       : XShortEnumField      = 2048            (0)\n--\nversion    : BitField             = 4               (4)\nihl        : BitField             = None            (None)\ntos        : XByteField           = 0               (0)\nlen        : ShortField           = None            (None)\nid         : ShortField           = 108             (1)\nflags      : FlagsField           = 0               (0)\nfrag       : BitField             = 0               (0)\nttl        : ByteField            = 63              (64)\nproto      : ByteEnumField        = 6               (0)\nchksum     : XShortField          = None            (None)\nsrc        : Emph                 = u'100.1.1.1'    (None)\ndst        : Emph                 = u'30.1.10.102'  ('127.0.0.1')\noptions    : PacketListField      = []              ([])\n--\nsport      : ShortEnumField       = 1234            (20)\ndport      : ShortEnumField       = 5000            (80)\nseq        : IntField             = 0               (0)\nack        : IntField             = 0               (0)\ndataofs    : BitField             = None            (None)\nreserved   : BitField             = 0               (0)\nflags      : FlagsField           = 2               (2)\nwindow     : ShortField           = 8192            (8192)\nchksum     : XShortField          = None            (None)\nurgptr     : ShortField           = 0               (0)\noptions    : TCPOptionsField      = {}              ({})\n--\nload       : StrField             = '\\x00\\x01\\x02\\x03\\x04\\x05\\x06\\x07\\x08\\t\\n\\x0b\\x0c\\r\\x0e\\x0f\\x10\\x11\\x12\\x13\\x14\\x15\\x16\\x17\\x18\\x19\\x1a\\x1b\\x1c\\x1d\\x1e\\x1f !\"#$%&\\'()*+,-' ('')\n--\n0000   24 8A 07 AB 9F 02 98 03  9B F3 F5 80 81 00 0B B9   $...............\n0010   08 00 45 00 00 56 00 6C  00 00 3F 06 ED CD 64 01   ..E..V.l..?...d.\n0020   01 01 1E 01 0A 66 04 D2  13 88 00 00 00 00 00 00   .....f..........\n0030   00 00 50 02 20 00 ED DE  00 00 00 01 02 03 04 05   ..P. ...........\n0040   06 07 08 09 0A 0B 0C 0D  0E 0F 10 11 12 13 14 15   ................\n0050   16 17 18 19 1A 1B 1C 1D  1E 1F 20 21 22 23 24 25   .......... !\"#$%\n0060   26 27 28 29 2A 2B 2C 2D                            &'()*+,-\n========== RECEIVED ==========\n0 total packets.\n==============================\n\n\n----------------------------------------------------------------------\nRan 1 test in 3.027s\n\nFAILED (failures=1)",

### CODE REVIEW

File: ansible/roles/test/files/ptftests/vnet_vxlan.py [hunk 1]
```diff
@@ -300,11 +301,21 @@ def setUp(self):
         self.generate_ArpResponderConfig()
 
         self.cmd(["supervisorctl", "start", "arp_responder"])
-
+        self.check_arp_responder_running()
         self.dataplane.flush()
 
         return
 
+    def check_arp_responder_running(self):
```
Comment (roysr-nv) [inline ansible/roles/test/files/ptftests/vnet_vxlan.py:309]: what if the arp responder is not up after 5 seconds - in this case I don't see any exception thrown.

Have you considered using the retry library or retry decorator?

Comment (andriyz-nv) [reply ansible/roles/test/files/ptftests/vnet_vxlan.py:309]: retry library or decorator is not installed to ptf docker.

**Comment by andriyz-nv** [comment]
@yxieca Can you please review?

File: ansible/roles/test/files/ptftests/vnet_vxlan.py [hunk 2]
```diff
@@ -300,11 +301,23 @@ def setUp(self):
...
+        """
+        Check arp_responder is in RUNNING state, if not sleep 1 sec and check again
+        """
+        for i in range(5):
+            output = self.cmd(["supervisorctl", "status", "arp_responder"])
+            if 'RUNNING' in output[0]:
+                break
+            time.sleep(1)
+        else:
+            raise Exception("arp_responder state is not RUNNING! Output: %s" % output)
```
Comment (yxieca) [inline ansible/roles/test/files/ptftests/vnet_vxlan.py:319]: Why not using wait_until?

Comment (andriyz-nv) [reply ansible/roles/test/files/ptftests/vnet_vxlan.py:319]: Hi @yxieca 

This is executed inside the  ptf docker, it does not has sonic_mgmt to do imports .
I did check the https://github.com/p4lang/ptf/ and did not find suitable

File: ansible/roles/test/files/ptftests/vnet_vxlan.py (hunk 1 above)
Comment (yxieca) [reply ansible/roles/test/files/ptftests/vnet_vxlan.py:309]: I remember there was an recent change to install retry in sonic-mgmt docker. But we do need to be careful here to make sure people using older sonic-mgmt docker would have a fair warning.

**Review by yxieca — APPROVED** [review]

**Review by roysr-nv — APPROVED** [review]

**Review by wangxin — APPROVED** [review]

**Pull request merged**